### PR TITLE
Added .dockerignore file and added dotnet test to Dockerfile

### LINKED ISSUE #5: Run Unit Tests when building a new image
Add a new stage in the Dockerfile in order to execute the Unit Tests at building time such as the Dockerfile will have three stages
1. Stage to build the app.
2. Stage to run Unit Tests and Publish the DLLs.
3. Stage to run the DLLs generated in the previous stage.

**Acceptance Criteria:**
- Get a new image which contains the DLLs.
- Implement a file `.dockerignore` properly.
- Make sure that your CI workflow is running with these changes with no errors.

**Deadline:**
Dec, 21th, 2022

### LINKED ISSUE #6: Build a new image using Docker Compose
Make sure that your VM has Docker Compose installed. Then, implement a `docker-compose.yaml`file which contains the code to build a new container image using the Multistage Dockerfile by using the command `docker-compose build`.

**Acceptance Criteria:**
- The file `docker-compose.yaml` should be created at the same place where the Dockerifle is.
- Build a new container image in your CI workflow by using the command `docker-compose build`

**Deadline:**
Dec, 21th, 2022

### PR DESCRIPTION
**Pull request by adrielgamorim**
- Added .dockerignore file to root folder of project;
- Added a new stage in the Dockerfile to run `dotnet build` and `dotnet test`;
- Refactored kof.yml to work in the new directory;
- Added docker-compose.yml file.

resolves #5
resolves #6

### CODE REVIEW

**Review by gioacchinogioffre — APPROVED** [review]
Good job!

**Review by gioacchinogioffre — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build-env
+WORKDIR /
+# Copy everything
+COPY . ./
+# Restore as distinct layers
+RUN cd Game && dotnet restore
+# Build and publish a release
+RUN cd Game && dotnet publish -c Release -o out
```
Comment (calebespinoza) [inline Dockerfile:17]: I think the better is to move this line into the second stage where you run the Unit Tests.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+RUN cd Game && dotnet build
+# Run all test cases
+RUN cd Game && dotnet test
+
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build-env
+WORKDIR /
+# Copy everything
+COPY . ./
+# Restore as distinct layers
+RUN cd Game && dotnet restore
```
Comment (calebespinoza) [inline Dockerfile:15]: Please, fix the correct folder name, you should enter into the `GameTest` folder to run the command `dotnet test`

**Review by Luisjor — APPROVED** [review]

**Review by calebespinoza — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (adrielgamorim) [reply Dockerfile:17]: Done.

File: Dockerfile (hunk 2 above)
Comment (adrielgamorim) [reply Dockerfile:15]: Done

**Pull request merged**